portal-impl/src/com/liferay/portlet/internal/PortletRequestImpl.java

```diff
@@ -1155,17 +1333,64 @@ public boolean isNameInvalid(boolean strutsPortlet) {
 			return false;
 		}
 
-		// TODO
+		public boolean isPrivateRenderNamespaced() {
+			return _privateRenderNamespaced;
+		}
+
+		public void setValues(String[] values) {
+			_values = values;
+		}
+
+		private static String _getName(String name) {
+			if (name != null) {
+				int pos = name.indexOf(
+					PortletQName.PRIVATE_RENDER_PARAMETER_NAMESPACE);
+
+				int privateRenderParameterNamespaceLength =
+					PortletQName.PRIVATE_RENDER_PARAMETER_NAMESPACE.length();
+
+				if (pos >= 0) {
+					name = name.substring(
+						0,
+						pos).concat(
+							name.substring(
+								pos + privateRenderParameterNamespaceLength));
+				}
+			}
+
+			return name;
+		}
 
 		private RequestParameter(
-			String name, String[] values, String portletNamespace) {
+			String name, String[] values, String portletNamespace,
+			int portletSpecMajorVersion) {
+
+			super(_getName(name), values, portletNamespace);
```

Comment:
@shuyangzhou In a prior pull request, you asked me to create a `private static _getName(String name)` method that would change the value of the name parameter, and then call `super`. I was able to do that for the name parameter, but I was not able to do that for the `values` parameter.